### PR TITLE
Wrapper function for Pagtn Model

### PR DESCRIPTION
**Pull request by VIGNESHinZONE**
Building the wrapper function for PAGTN model https://github.com/deepchem/deepchem/issues/2389
The model and featurisers are already built-in DGL-Lifesci [#138](https://github.com/awslabs/dgl-lifesci/pull/138) [#139](https://github.com/awslabs/dgl-lifesci/pull/139) and now needs a wrapping function.

In this PR I'm adding - 
1) Wrapper function for the Pagtn Model from DGL Lifesci
2) Tests for classification, regression, and reload

I have maintained the same style as that kept by other Torch Graph Models -  `AttentiveFPModel` and `GatModel`

cc: @mufeili  as it is a Wrapper Function from DGL-LifeSci ModelZoo

### CODE REVIEW

File: deepchem/models/torch_models/pagtn.py [hunk 1]
```diff
@@ -0,0 +1,306 @@
...
+  """Model for Graph Property Prediction
+
+  This model proceeds as follows:
+
+  * Update node representations in graphs with a variant of GAT, where a
+    linear additive form of attention is applied. Attention Weights are derived
+    by concatenating the node and edge features for each bond.
+  * Update node representations with multiple rounds of message passing.
+  * For each layer has, residual connections with its previous layer and finally compute its
+    representation by combining the representations of all nodes in it.
```
Comment (mufeili) [inline deepchem/models/torch_models/pagtn.py:21]: Maybe you want to break L20-21 into two sentences, i.e.
```python
"""
* Each layer has residual connections with its previous layer.
* The final molecular representation is computed by combining the representations of all nodes in the molecule.
"""
```

File: deepchem/models/torch_models/pagtn.py [hunk 2]
```diff
@@ -0,0 +1,306 @@
...
+  (https://github.com/awslabs/dgl-lifesci) to be installed.
+  """
+
+  def __init__(self,
+               n_tasks: int,
+               number_atom_features: int = 94,
+               number_bond_features: int = 42,
+               mode: str = 'regression',
+               n_classes: int = 2,
+               ouput_node_features: int = 256,
```
Comment (mufeili) [inline deepchem/models/torch_models/pagtn.py:61]: "ouput" -> "output"?

File: deepchem/models/torch_models/pagtn.py [hunk 3]
```diff
@@ -0,0 +1,306 @@
...
+    number_atom_features : int
+      Size for the input node features. Default to 94.
+    number_bond_features : int
+      Size for the input edge features. Default to 42.
+    mode: str
+      The model type, 'classification' or 'regression'. Default to 'regression'.
+    n_classes: int
+      The number of classes to predict per task
+      (only used when ``mode`` is 'classification'). Default to 2.
+    ouput_node_features : int
```
Comment (mufeili) [inline deepchem/models/torch_models/pagtn.py:83]: Ditto

File: deepchem/models/torch_models/pagtn.py [hunk 4]
```diff
@@ -0,0 +1,306 @@
...
+  """Model for Graph Property Prediction.
+
+  This model proceeds as follows:
+
+  * Update node representations in graphs with a variant of GAT, where a
+    linear additive form of attention is applied. Attention Weights are derived
+    by concatenating the node and edge features for each bond.
+  * Update node representations with multiple rounds of message passing.
+  * For each layer has, residual connections with its previous layer and finally compute its
+    representation by combining the representations of all nodes in it.
```
Comment (mufeili) [inline deepchem/models/torch_models/pagtn.py:193]: Ditto

File: deepchem/models/torch_models/pagtn.py [hunk 5]
```diff
@@ -0,0 +1,306 @@
...
+  (https://github.com/awslabs/dgl-lifesci) to be installed.
+  """
+
+  def __init__(self,
+               n_tasks: int,
+               number_atom_features: int = 94,
+               number_bond_features: int = 42,
+               mode: str = 'regression',
+               n_classes: int = 2,
+               num_layers: int = 5,
```
Comment (mufeili) [inline deepchem/models/torch_models/pagtn.py:227]: Seems we are missing `ouput_node_features` and `hidden_features`

File: deepchem/models/torch_models/pagtn.py [hunk 6]
```diff
@@ -0,0 +1,306 @@
...
+      loss: Loss = L2Loss()
+      output_types = ['prediction']
+    else:
+      loss = SparseSoftmaxCrossEntropy()
+      output_types = ['prediction', 'loss']
+    super(PagtnModel, self).__init__(
+        model, loss=loss, output_types=output_types, **kwargs)
+
+  def _prepare_batch(self, batch):
+    """Create batch data for AttentiveFP.
```
Comment (mufeili) [inline deepchem/models/torch_models/pagtn.py:278]: "AttentiveFP" -> "PAGTN"

**Comment by mufeili** [comment]
Done a first pass.

**Review by rbharath — COMMENTED** [review]
I think this basically looks ready to merge in! It would be really nice if we could figure out the dgllife installation issues we're facing before merging in if possible though

**Comment by rbharath** [comment]
@VIGNESHinZONE Now that #2516 is merged in with a dgl-lifesci fix, can you rebase this PR so we can check that everything works on the fixed CI? I'm hoping we can get this merged in by tomorrow so we can get it into 2.6.0 :)

**Comment by VIGNESHinZONE** [comment]
@rbharath 

I have fixed the https://github.com/deepchem/deepchem/issues/2517 changing the `maximum` and `multiply` method.

Github Actions checks have passed for Ubuntu 3.6 and 3.7 , I think its ready to merge

**Comment by rbharath** [comment]
Going to go ahead and merge this in for the release!

**Pull request merged**